src/interfaces/node.cpp

```diff
@@ -270,7 +270,7 @@ class NodeImpl : public Node
     }
     std::unique_ptr<Handler> handleLoadWallet(LoadWalletFn fn) override
     {
-        return MakeHandler(::uiInterface.LoadWallet_connect([fn](std::shared_ptr<CWallet> wallet) { fn(MakeWallet(wallet)); }));
+        return MakeHandler(::uiInterface.LoadWallet_connect([fn](std::unique_ptr<Wallet>& wallet) { fn(std::move(wallet)); }));
```

Comment:
It's not obvious to me why this (and associated changes) is necessary. Can you perhaps add commentary to the commit message why you're changing the function signature here?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15288#discussion_r254464908

> It's not obvious to me why this (and associated changes) is necessary

I'll add information to the commit message, but just to give some feedback now, the idea is that in #10102 the `CWallet` class is internal to the `bitcoin-wallet` process. The other `bitcoin-node` and `bitcoin-gui` processes don't use `CWallet` at all, and only access the wallet through interface classes (`interfaces::ChainClient` and `interfaces::Wallet` respectively).

The change from `shared_ptr` to `unique_ptr` here just follows because `CWallet` is used with `shared_ptr`, while the interface classes have more defined lifetimes and don't need `shared_ptr`. Any place an interface argument or return value is passed it's always by reference or with `unique_ptr`.